src/tests/test_faida.cpp

```diff
@@ -0,0 +1,213 @@
+            expected_subset.erase(iter);
+        }
+    }
+    ASSERT_EQ(expected_subset.size(), 0);
+}
+
+// Since Faida is an approximate algorithm, its results may differ from all the other IND discovery
+// algorithms, which are exact. Of course, on these small examples the results of both exact and
+// approximate algos must be the same, but in the future the set of test datasets will grow and
+// large tests may be added. Therefore the decision was made to move Faida tests to a separate
+// class.
+class FaidaINDAlgorithmTest : public ::testing::Test {
+protected:
+    static algos::StdParamsMap GetParamMap(std::vector<fs::path> const& paths, char separator = ',',
+                                           bool has_header = true, int sample_size = 500,
+                                           double hll_accuracy = 0.001, bool find_nary = true,
+                                           unsigned short num_threads = 4) {
+        using namespace config::names;
+        return {{kCsvPaths, paths},     {kSeparator, separator},    {kHasHeader, has_header},
+                {kFindNary, find_nary}, {kSampleSize, sample_size}, {kHllAccuracy, hll_accuracy},
+                {kThreads, num_threads}};
+    }
+
+    template <typename... Args>
+    static std::unique_ptr<algos::INDAlgorithm> CreateFaidaInstance(Args&&... args) {
+        return algos::CreateAndLoadAlgorithm<algos::Faida>(
+                GetParamMap(std::forward<Args>(args)...));
+    }
+
+    static fs::path GetDataPath(std::string const& data_name) {
+        return test_data_dir / "ind_data" / data_name;
+    }
+};
+
+TEST_F(FaidaINDAlgorithmTest, TestWide2) {
+    IndsTest expected_inds{{{0, {2}}, {0, {0}}}, {{0, {3}}, {0, {1}}}, {{0, {2, 3}}, {0, {0, 1}}}};
+
+    auto file_name = "TestWide2.csv";
+    char separator = ',';
+    bool has_header = false;
+    int sample_size = 500;
+    double hll_accuracy = 0.001;
+    bool find_nary = true;
+    unsigned short num_threads = 4;
+
+    auto path = GetDataPath(file_name);
+
+    auto algorithm = CreateFaidaInstance(std::vector{path}, separator, has_header, sample_size,
+                                         hll_accuracy, find_nary, num_threads);
+    algorithm->Execute();
+    CheckINDsListsEquality(algorithm->INDList(), expected_inds);
+}
+
+TEST_F(FaidaINDAlgorithmTest, TestEmpty) {
+    IndsTest expected_inds{};
+
+    auto file_name = "Empty.csv";
+    char separator = ',';
+    bool has_header = true;
+    int sample_size = 500;
+    double hll_accuracy = 0.001;
+    bool find_nary = true;
+    unsigned short num_threads = 4;
+
+    auto path = GetDataPath(file_name);
+
+    ASSERT_THROW(CreateFaidaInstance(std::vector{path}, separator, has_header, sample_size,
+                                     hll_accuracy, find_nary, num_threads),
+                 std::runtime_error);
+}
+
+TEST_F(FaidaINDAlgorithmTest, TestEmptyFolder) {
+    IndsTest expected_inds{};
+
+    auto file_name = "empty_dataset";
```

Comment:
it is worth adding configs to the [all_tables_config.h](https://github.com/Mstrutov/Desbordante/blob/main/src/tests/all_tables_config.h) and using them in this file